packages/ember-application/tests/system/initializers_test.js

```diff
@@ -34,6 +34,42 @@ QUnit.test('initializers require proper \'name\' and \'initialize\' properties',
   });
 });
 
+if (isEnabled('ember-application-visit')) {
+  QUnit.test('initializers that thorws causes the boot promise to reject with the error', function() {
+    QUnit.expect(2);
+    QUnit.stop();
+
+    var MyApplication = Application.extend();
+
+    MyApplication.initializer({
+      name: 'initializer',
+      initialize() { throw new Error('boot failure'); }
+    });
+
+    var app = MyApplication.create({
+      autoboot: false
+    });
+
+    try {
+      app.boot().then(
```

Comment:
qunit supports promises, you can just `return app.boot()` or `return.app.boot().catch(...)` for a more concise test.